onyo/lib/onyo.py

```
@@ -92,6 +92,94 @@ def clear_caches(self) -> None:
+                                                                    dest, MAX_LEN)
+
+        message_appendix = self._n_join([str(x) for x in staged_changes])
+        return f"{message_subject}\n\n{message_body}\n\n{message_appendix}"
+
+    def _generate_commit_message_subject(self, msg_dummy: str,
+                                         staged_changes: list[Path],
+                                         destination: Optional[Path],
+                                         MAX_LEN: int) -> str:
+        """
+        Generates "commit message subject" with the `msg_dummy` and the paths
+        from `staged_changes` and `destination`, and shortens the paths if the
+        message length exceeds the `MAX_LEN`.
+        """
+
+        # long message: full paths (relative to repo-root)
+        paths_str = ','.join(["'{}'".format(x) for x in staged_changes])
+        msg = f"{msg_dummy}: {paths_str}"
+        if destination:
+            msg = f"{msg} -> '{destination}'"
+
+        if len(msg) < MAX_LEN:
+            return msg
+
+        # medium message: highest level (e.g. dir or asset name)
+        paths = [str(x.name) for x in staged_changes]
+        paths_str = ','.join(["'{}'".format(x) for x in paths])
+        msg = f"{msg_dummy}: {paths_str}"
+        if destination:
+            msg = f"{msg} -> '{destination.relative_to(destination.parent)}'"
+
+        if len(msg) < MAX_LEN:
+            return msg
+
+        # short message: "type" of devices in summary (e.g.  "laptop (2)")
+        paths = [str(x.name).split('_')[0] for x in staged_changes]
+        paths_str = ','.join(sorted(["'{} ({})'".format(x, paths.count(x))
+                                     for x in set(paths)]))
+        msg = f"{msg_dummy}: {paths_str}"
+        if destination:
+            msg = f"{msg} -> '{destination.relative_to(destination.parent)}'"
+
+        # return the shortest possible version of the commit message as default
```

I think "default" is misleading here. It's rather a "fallback".